src/logic/add.rs

```diff
@@ -0,0 +1,265 @@
+    pub fn wrapping_dec(&mut self) {
+        match self.access_data_mut() {
+            DataAccessMut::Inl(x) => {
+                *x = x.wrapping_sub(Digit::one());
+            }
+            DataAccessMut::Ext(x) => {
+                for i in 0..x.len() {
+                    match x[i].overflowing_sub(Digit::one()) {
+                        (v,false) => {
+                            x[i] = v;
+                            break;
+                        }
+                        (v,true) => {
+                            //if the ApInt was relatively random this should rarely happen
+                            x[i] = v;
+                        }
+                    }
+                }
+            }
+        }
+        self.clear_unused_bits();
+    }
+
+    /// Decrements this `ApInt` by one and returns the result.
+    pub fn into_wrapping_dec(self) -> ApInt {
+        forward_mut_impl(self, ApInt::wrapping_dec)
+    }
+
+    /// Negates this `ApInt` inplace.
+    pub fn wrapping_neg(&mut self) {
+        self.bitnot();
+        self.wrapping_inc();
+        //`wrapping_inc` handles clearing the unused bits
+    }
+
+    /// Negates this `ApInt` and returns the result.
+    pub fn into_wrapping_neg(self) -> ApInt {
+        forward_mut_impl(self, ApInt::wrapping_neg)
+    }
+
+    /// Add-assigns `rhs` to `self` inplace.
+    pub fn wrapping_add_assign(&mut self, rhs: &ApInt) -> Result<()> {
+        match self.zip_access_data_mut_self(rhs)? {
+            Inl(lhs, rhs) => {
+                *lhs = lhs.wrapping_add(rhs);
+            }
+            Ext(lhs, rhs) => {
+                let (temp, mut carry) = lhs[0].carrying_add(rhs[0]);
+                lhs[0] = temp;
+                for i in 1..lhs.len() {
+                    let temp = lhs[i].dd()
+                        .wrapping_add(rhs[i].dd())
+                        .wrapping_add(carry.dd());
+                    lhs[i] = temp.lo();
+                    carry = temp.hi();
+                }
+            }
+        }
+        self.clear_unused_bits();
+        Ok(())
+    }
+
+    /// Adds `rhs` to `self` and returns the result.
+    pub fn into_wrapping_add(self, rhs: &ApInt) -> Result<ApInt> {
+        try_forward_bin_mut_impl(self, rhs, ApInt::wrapping_add_assign)
+    }
+
+    /// Add-assigns `rhs` to `self` inplace, and returns a boolean indicating if overflow occured,
+    /// according to the **unsigned** interpretation of overflow.
+    // TODO: add tests
+    #[allow(dead_code)]
+    pub(crate) fn overflowing_uadd_assign(&mut self, rhs: &ApInt) -> Result<bool> {
+        match self.width().excess_bits() {
+            Some(excess) => {
+                let mask = Digit::all_set() >> excess;
+                match self.zip_access_data_mut_self(rhs)? {
+                    Inl(lhs, rhs) => {
+                        let temp = lhs.wrapping_add(rhs);
+                        *lhs = temp & mask;
+                        //excess bits are cleared by the mask
```

Comment:
```suggestion
                        // Excess bits are cleared by the mask.
```

Reply:
I think by now you might get the idea that I am really eager about this whitespace behind the `//` and the big letter afterwards and the general punctuation. I just find it to be looking better, more readable and thus increasing the likelyhood of other programmers to do higher quality comments as well. That's the reason. So if I overlooked some malformatted non-doc comments, please fix them as well!